src/redux/books/books.js

```diff
@@ -44,18 +18,66 @@ export const removeBook = (id) => ({
   id,
 });
 
+const setBooks = (payload) => ({
+  type: SET_BOOKS,
+  payload,
+});
+
+export const createBook = (book) => async (dispatch) => {
+  const isCreated = await API.createBook(book);
+  if (isCreated) {
+    dispatch(addBook(book));
+  }
+};
+
+export const deleteBook = (id) => async (dispatch) => {
+  const isDeleted = await API.deleteBook(id);
+
+  if (isDeleted) {
+    dispatch(removeBook(id));
+  }
+};
+
+export const loadBooks = () => async (dispatch) => {
+  const books = await API.getBooks();
+
+  if (books) {
+    dispatch(setBooks(books));
+  }
+};
+
 const reducer = (state = initialState, action) => {
   switch (action.type) {
     case ADD_BOOK:
-      return [...state, {
-        ...action.payload,
-        progress: {
-          currentChapter: 'Preface',
-          completed: '0',
+      return [
+        ...state,
+        {
+          ...action.payload,
+          author: 'Author not set',
+          progress: {
+            currentChapter: 'Introduction',
+            completed: '0',
+          },
         },
-      }];
+      ];
+    case SET_BOOKS: {
+      const saved = Object.entries(action.payload).map(([key, value]) => {
+        const [book] = value;
```

Comment:
- [ ] Please fix this error. And remember that the data must persist all changes in the server which the next reviewer will look after you fix this error
![Screenshot from 2021-11-03 23-31-58](https://user-images.githubusercontent.com/62705947/140209848-ca168f9a-a57b-45c5-a5e9-65883944942f.png)

Reply:
This error is now fixed